test/active_test.go

```diff
@@ -74,19 +72,8 @@ var _ = Describe("pkg/active with apiserver", func() {
 		}
 		err := c.Delete(context.Background(), ns)
 		Expect(err).NotTo(HaveOccurred())
-		// Validate the calico-system namespace is deleted using an unstructured type.
-		// This hits the API server directly instead of using the client cache.
-		// This should help with flaky tests.
 		Eventually(func() error {
-			u := &unstructured.Unstructured{}
-			u.SetGroupVersionKind(schema.GroupVersionKind{
-				Group:   "",
-				Version: "v1",
-				Kind:    "Namespace",
-			})
-
-			k := client.ObjectKey{Name: "calico-system"}
-			err := c.Get(context.Background(), k, u)
+			err := GetResource(c, ns)
```

Comment:
nit: you could just do `return GetResource(c, ns)`

Reply:
Took it once step further and removed the func definition entirely so value of `GetResource` could be evaluated directly.